hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/viewfs/ViewFs.java

```diff
@@ -973,18 +976,27 @@ static class InternalDirOfViewFs extends AbstractFileSystem {
     final URI myUri; // the URI of the outer ViewFs
     private InodeTree<AbstractFileSystem> fsState;
     private Configuration conf;
+    private String mountLinkUserName;
+    private String mountLinkGroupName;
 
     public InternalDirOfViewFs(final InodeTree.INodeDir<AbstractFileSystem> dir,
         final long cTime, final UserGroupInformation ugi, final URI uri,
-        InodeTree fsState, Configuration conf)
-      throws URISyntaxException {
+        InodeTree fsState, Configuration conf) throws URISyntaxException, IOException {
       super(FsConstants.VIEWFS_URI, FsConstants.VIEWFS_SCHEME, false, -1);
       theInternalDir = dir;
       creationTime = cTime;
       this.ugi = ugi;
       myUri = uri;
       this.fsState = fsState;
       this.conf = conf;
+      mountLinkUserName = conf.get(CONFIG_VIEWFS_MOUNT_LINKS_USER_NAME);
```

Comment:
may be write this as conf.get(CONFIG_VIEWFS_MOUNT_LINKS_USER_NAME) == null ? ugi.getShortUserName() :  conf.get(CONFIG_VIEWFS_MOUNT_LINKS_USER_NAME) and explain why you are not using  conf.get(CONFIG_VIEWFS_MOUNT_LINKS_USER_NAME, ugi.getShortUserName()) in a comment? Otherwise, it's unclear why you aren't using that method directly. Same for the other places where you are using this template